soh/src/overlays/actors/ovl_En_Ds/z_en_ds.c

```diff
@@ -195,17 +196,19 @@ void EnDs_OfferBluePotion(EnDs* this, PlayState* play) {
                         this->actionFunc = EnDs_TalkNoEmptyBottle;
                         return;
                     case 2: // have 100 rupees and empty bottle
-                        Rupees_ChangeBy(-100);
+                        if(!GameInteractor_Should(VB_RANDO_GRANNY_TAKE_MONEY, false, this)){
+                            Rupees_ChangeBy(-100);
+                        }
                         this->actor.flags &= ~ACTOR_FLAG_WILL_TALK;
 
-                        if (GameInteractor_Should(VB_GIVE_ITEM_FROM_GRANNYS_SHOP, true, this)) {
+                        if (!GameInteractor_Should(VB_GIVE_ITEM_FROM_GRANNYS_SHOP, false, this)) {
```

Comment:
Same as with above, would probably be better removing the double inversion.
```suggestion
                        if (GameInteractor_Should(VB_GIVE_ITEM_FROM_GRANNYS_SHOP, true, this)) {
```